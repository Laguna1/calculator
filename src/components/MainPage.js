import Calculator from './Calculator';

const MainPage = () => (
  <div className="">
    <h2>Hi there, User!</h2>
    <div>
      <Calculator />
    </div>
  </div>
);

export default MainPage;
