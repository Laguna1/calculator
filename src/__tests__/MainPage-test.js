import renderer from 'react-test-renderer';
import {
  render, fireEvent, waitFor, screen,
} from '@testing-library/react';
import MainPage from '../components/MainPage';

describe('MainPage', () => {
  it('renders correctly', () => {
    const page = renderer.create(<MainPage />);
    expect(page).toMatchSnapshot();
  });

  it('Display shows the clicked number', async () => {
    render(<MainPage />);
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('2'));
    const res = await waitFor(() => screen.findAllByText('12'));
    expect(res).toHaveLength(1);
  });

  it('Caculator shows a zero upon launch', async () => {
    render(<MainPage />);
    const res = await waitFor(() => screen.findAllByText('0'));
    expect(res).toHaveLength(2);
  });

  it('Displays sum on addition of two numbers', async () => {
    render(<MainPage />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    const res = await waitFor(() => screen.findAllByText('12'));
    expect(res).toHaveLength(1);
  });

  it('Displays product on multiplying of two numbers', async () => {
    render(<MainPage />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    const res = await waitFor(() => screen.findAllByText('35'));
    expect(res).toHaveLength(1);
  });

  it('Changes the sign of the displayed figure when "+/-" is clicked', async () => {
    render(<MainPage />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('+/-'));
    const res = await waitFor(() => screen.findAllByText('-57'));
    expect(res).toHaveLength(1);
  });

  it('Displays the percentantage of figure when "%" is clicked', async () => {
    render(<MainPage />);
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('%'));
    const newLocal = await waitFor(() => screen.findAllByText('0.05'));
    const res = newLocal;
    expect(res).toHaveLength(1);
  });
});
