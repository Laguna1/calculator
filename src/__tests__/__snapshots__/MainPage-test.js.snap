// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="sc-bdfBwQ kHSvhl"
  >
    <div
      className="display"
    >
      0
    </div>
    <div>
      <div
        className="btn-row"
      >
        <button
          className="button grey"
          name="AC"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="button grey"
          name="+/-"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="button grey"
          name="%"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="button orange"
          name="/"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="/"
        >
          /
        </button>
      </div>
      <div
        className="btn-row"
      >
        <button
          className="button grey"
          name="7"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="button grey"
          name="8"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="button grey"
          name="9"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="button orange"
          name="x"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="x"
        >
          x
        </button>
      </div>
      <div
        className="btn-row"
      >
        <button
          className="button grey"
          name="4"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="button grey"
          name="5"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="button grey"
          name="6"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="button orange"
          name="-"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="-"
        >
          -
        </button>
      </div>
      <div
        className="btn-row"
      >
        <button
          className="button grey"
          name="1"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="button grey"
          name="2"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="button grey"
          name="3"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="button orange"
          name="+"
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="+"
        >
          +
        </button>
      </div>
      <div
        className="btn-row"
      >
        <button
          className="button grey"
          name="0"
          onClick={[Function]}
          style={
            Object {
              "width": "50%",
            }
          }
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="button grey"
          name="."
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="button orange"
          name="="
          onClick={[Function]}
          style={
            Object {
              "width": "25%",
            }
          }
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
